test/position-managers/FeeCollection.t.sol

```diff
@@ -136,6 +137,107 @@ contract FeeCollectionTest is Test, PosmTestSetup, LiquidityFuzzers {
         assertEq(manager.balanceOf(address(lpm), currency1.toId()), 0);
     }
 
+    function test_collect_donate() public {
+        LiquidityRange memory range = LiquidityRange({poolKey: key, tickLower: -120, tickUpper: 120});
+        mint(range, 10e18, address(this), ZERO_BYTES);
+        uint256 tokenId = lpm.nextTokenId() - 1;
+
+        // donate to generate fee revenue
+        uint256 feeRevenue = 1e18;
+        donateRouter.donate(key, feeRevenue, feeRevenue, ZERO_BYTES);
+
+        BalanceDelta expectedFees = IPositionManager(address(lpm)).getFeesOwed(manager, tokenId);
+
+        // collect fees
+        uint256 balance0Before = currency0.balanceOfSelf();
+        uint256 balance1Before = currency1.balanceOfSelf();
+        BalanceDelta delta = collect(tokenId, ZERO_BYTES);
+
+        assertApproxEqAbs(uint256(int256(delta.amount0())), feeRevenue, 1 wei);
+        assertApproxEqAbs(uint256(int256(delta.amount1())), feeRevenue, 1 wei);
+        assertEq(delta.amount0(), expectedFees.amount0());
+        assertEq(delta.amount1(), expectedFees.amount1());
+
+        assertEq(uint256(int256(delta.amount0())), currency0.balanceOfSelf() - balance0Before);
```

Comment:
i feel like its more intuitive to me that we assert the balanceBefore + expectedFees equals the balance now